pkg/kubelet/container/helpers.go

```diff
@@ -93,6 +96,16 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 // the running container with its desired spec.
 func HashContainer(container *v1.Container) uint64 {
 	hash := fnv.New32a()
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		cResources := container.Resources
```

Comment:
Do we need to exclude ResizePolicy as well?

Reply:
No, we don't. At this time, we don't allow user to change resize policy once a pod has been started. Only Resources and ResourcesAllocated fields becoming mutable shouldn't be allowed to change the hash that is used to determine container needs restart (e.g. image change must result in restart, but limit change should only trigger update)